test_aodndata/aatams/test_aatams_sattag.py

```diff
@@ -0,0 +1,135 @@
+"""Tests for the AATAMS DM and NRT streams"""
+import os
+import unittest
+from testfixtures import LogCapture
+from aodncore.pipeline import FileType, PipelineFilePublishType
+from aodncore.pipeline.exceptions import InvalidInputFileError, InvalidFileContentError
+from aodncore.testlib import HandlerTestCase
+from aodndata.aatams.aatams_sattag import (
+    AatamsSattagQcNRTHandler,
+    AatamsSattagQcDmHandler,
+    AATAMS_SATTAG_QC_DM_BASE,
+    AATAMS_SATTAG_QC_NRT_BASE,
+    NRT_TIMESTAMP_COMPARISON_MSG,
+)
+
+TEST_ROOT = os.path.join(os.path.dirname(__file__))
+
+DM_GOOD_ZIP = os.path.join(TEST_ROOT, "simple_dm.zip")
+DM_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_dm.csv")
+
+NRT_FIRST_ZIP = os.path.join(TEST_ROOT, "first_nrt.zip")
+NRT_SECOND_ZIP = os.path.join(TEST_ROOT, "second_nrt.zip")
+NRT_SINGLE_CSV = os.path.join(TEST_ROOT, "metadata_nrt.csv")
+
+
+def check_file(cls, file):
+    """Check if a file is a valid AATAMS file with correct storage options"""
+    if file.file_type is FileType.CSV:
+        cls.assertEqual(
+            cls.base_path, os.path.dirname(file.dest_path),
+        )
+        cls.assertTrue(file.is_harvested and file.is_stored)
+    elif file.file_type is FileType.ZIP:
+        cls.assertEqual(
+            cls.base_path, os.path.dirname(file.archive_path),
+        )
+        cls.assertTrue(file.is_archived)
+    else:
+        raise ValueError("File %s not a Zip or CSV" % file)
+
+
+class TestAatamsDmHandler(HandlerTestCase):
+    """ Tests for the DM pipeline """
+    base_path = AATAMS_SATTAG_QC_DM_BASE
+    check_file = check_file
+
+    def setUp(self):
+        self.handler_class = AatamsSattagQcDmHandler
+        super(TestAatamsDmHandler, self).setUp()
+
+    def test_avoid_nrt_pickup(self):
+        self.run_handler_with_exception(InvalidInputFileError, NRT_FIRST_ZIP)
```

Comment:
is NRT_FIRST_ZIP a bad zip ??

Reply:
Yes, the NRT ZIP is a bad file for the DM harverster since it got a ```_nrt``` string. In practice this is the only problem with it though

Reply:
sorry I'm not sure I understand your sentence.

I thought that the good incoming zip files were named "something_nrt.zip" and "something_dm.zip".

when you say DM harvester, you mean pipeline?


>  In practice this is the only problem with it though

Sorry I don't get this

Reply:
>I thought that the good incoming zip files were named "something_nrt.zip" and "something_dm.zip".

Yes, ```something_nrt.zip``` (```something_dm.zip```) is valid for the NRT (DM) handler, but invalid for the DM (NRT) handler.

>when you say DM harvester, you mean pipeline?

Yeap, I meant harvester.

>In practice this is the only problem with it though

It was just a remark. The file contents (not their names) are actually valid for both dm/nrt because they share the same schema.

Reply:
ohhhh ok, gotcha!